src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
-public class MyHashMap<K, V> implements MyMap<K, V> {
+public class MyHashMap<K,V> implements MyMap<K,V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node[] table;
+    private int size;
+    private int capacity;
+    private int threshold;
+
+    {
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+        table = new Node[capacity];
+    }
+
+    private class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        private Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+            this.hash = calculateHashCode();
+        }
+
+        private int calculateHashCode() {
+            if (key != null) {
+                return Math.abs(key.hashCode() % capacity);
+            }
+            return 0;
+        }
```

Comment:
this method doesn't calculate hashcode, it calculates index in the map
so we should move this method to the top-level class and don't save the result of this method to hash variable